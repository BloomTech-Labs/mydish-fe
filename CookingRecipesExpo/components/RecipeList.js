import React, {useEffect} from 'react';
import Recipe from './Recipe';
import {ScrollView, View} from 'react-native';


const RecipeList = (props) => {
    let imageHeight = 0;
    let LeftHeight = 0;
    let RightHeight= 0;

    const [recipes, setRecipes] = React.useState([]);

    useEffect(() =>{
        setRecipes(props.props)

        // console.log("HELLLOOOOO",Math.floor(recipes.length/2)+1)
    },[]);
    
    const divideArray =()=>{
        if(Math.floor(recipes.length/2)  ==0){
            return 1
        }else{
            return Math.floor(recipes.length/2) 
        }      
    }
    
    const LeftHeightAdjustment = () => {    
        if(LeftHeight === 0){  
            LeftHeight= LeftHeight +1
            return 220
        }if(LeftHeight===1){
            LeftHeight = LeftHeight -1
            return 230
        }
    }
    const RightHeightAdjustment = () => {     
        if(RightHeight === 2){  
            RightHeight= RightHeight +1
            return 230
        }if(RightHeight===3){
            RightHeight = RightHeight -1
            return 220
        }
    }
    const adjustImageHeight = () => {  
        if(imageHeight === 0){  
            imageHeight= imageHeight +1
            return 150
        }if(imageHeight===1){
            imageHeight = imageHeight -1
            return 185
        }
    }

    return (
        <ScrollView>
            <View style={{flexDirection: 'row'}}>
                <View style={{flexDirection: 'column', marginLeft: "5%" }}>
                 {recipes.slice(0, divideArray()).map( recp =>  <Recipe key={recp.id} recipe={recp} imageHeight={adjustImageHeight()} cardHeight={LeftHeightAdjustment()}/>)}
                </View>
                 <View style={{flexDirection: 'column', marginRight: "150%"}}>
                 {recipes.slice(divideArray(), recipes.length+1).map( recp =>  <Recipe key={recp.id} recipe={recp} imageHeight={adjustImageHeight()} cardHeight={RightHeightAdjustment()}/>)}
                </View>
            </View>
         </ScrollView>
    )  
}

export default RecipeList;