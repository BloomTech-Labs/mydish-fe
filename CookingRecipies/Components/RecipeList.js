import React from 'react';
import styles from '../styles/styles';
import Recipe from './Recipe';
import {AppRegistry, View} from 'react-native';
// import {name as appName} from '../app.json';
import App from '../App';

const RecipeList = () => {

    return (
        <>
        <View/>
            <Recipe/>
         <View/>
         </>
    )  
}

// AppRegistry.registerComponent(appName, () => App );

export default RecipeList;

