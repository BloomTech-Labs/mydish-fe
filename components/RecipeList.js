import React from 'react';
import { useSelector } from 'react-redux';
import Recipe from './Recipe';
import { View, FlatList } from 'react-native';
import FancySpinner from './FancySpinner';

const RecipeList = ({ parent, folder }) => {
  const isLoading = useSelector((store) => store.cookbook.isLoading);

  // If the RecipeList is being rendered from the cookbook,
  //     grab the props that the cookbook is passing down.
  // If the RecipeList ISN'T coming from the cookbook, then we
  //     useSelector to get all of the recipes.
  const recipeList =
    parent === 'cookbook'
      ? folder
      : useSelector((store) => store.allRecipes.recipeList);

  if (isLoading) return <FancySpinner />;

  return (
    <View style={{ flexDirection: 'row', flexWrap: 'wrap' }}>
      {recipeList.length !== 0 && (
        <FlatList
          contentContainerStyle={{ paddingBottom: 150 }}
          data={recipeList}
          numColumns={2}
          keyExtractor={(item) => item.id.toString()}
          renderItem={({ item }) => <Recipe key={item.id} recipe={item} />}
        />
      )}
    </View>
  );
};

export default RecipeList;
