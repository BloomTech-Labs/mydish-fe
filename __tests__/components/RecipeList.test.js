import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../../components/RecipeList';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

describe('<RecipeList />', () => {
  it('matches snapshot', () => {
    // The RecipeList component needs a few things from the store,
    //     so we create a simple store for those things:
    const store = createStore(() => {
      return {
        allRecipes: { recipeList: [] },
        cookbook: { cookbookRecipes: [] },
      };
    });

    // Then we create our component tree with the provider:
    const renderTree = (
      <Provider store={store}>
        <RecipeList />
      </Provider>
    );
    const tree = renderer.create(renderTree).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
