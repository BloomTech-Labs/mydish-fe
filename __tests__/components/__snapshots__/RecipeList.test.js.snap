// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> matches snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
/>
`;
